godtools/App/Features/ToolsFilter/Presentation/ToolFilterLanguageSelectionView/ToolFilterLanguageSelectionViewModel.swift

```diff
@@ -50,7 +48,21 @@ class ToolFilterLanguageSelectionViewModel: ObservableObject {
         $appLanguage.eraseToAnyPublisher()
             .flatMap { appLanguage in
                 
-                return viewToolFilterLanguagesUseCase.viewPublisher(filteredByCategoryId: selectedCategory.id, translatedInAppLanguage: appLanguage)
+                return getUserToolFiltersUseCase.getUserToolFiltersPublisher(translatedInAppLanguage: appLanguage)
+                    .eraseToAnyPublisher()
+            }
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] userFilters in
+            
+                self?.selectedLanguage = userFilters.languageFilter
+                self?.selectedCategory = userFilters.categoryFilter
```

Comment:
Hey @rachaelblue I do see the selectedCategory is declared as mutable and is being changed here, so there is the potential that flatMap is called before selectedCategory is set.

Reply:
Good point.  Thanks!